### PR TITLE
Add Course details to registration and apprenticeship

### PR DESCRIPTION
**Pull request by benagain**
🦄

### CODE REVIEW

File: src/SFA.DAS.ApprenticeCommitments.Api.AcceptanceTests/Features/VerifyRegistrationSteps.cs [hunk 1]
```diff
@@ -127,10 +127,21 @@ public void ThenAnApprenticeshipRecordIsCreated()
...
+                    _registration.Apprenticeship.TrainingProviderId,
+                    _registration.Apprenticeship.TrainingProviderName,
+                    Course = new
+                    {
+                        _registration.Apprenticeship.Course.Name,
+                        _registration.Apprenticeship.Course.Level,
+                        _registration.Apprenticeship.Course.Option,
+                        _registration.Apprenticeship.Course.PlannedStartDate,
+                        _registration.Apprenticeship.Course.PlannedEndDate,
+                    }
```
Comment (Pauljgraham) [inline src/SFA.DAS.ApprenticeCommitments.Api.AcceptanceTests/Features/VerifyRegistrationSteps.cs:143]: I like this pattern. Keeps entities together and we can use Value objects

File: src/SFA.DAS.ApprenticeCommitments/Data/Models/ApprenticeCommitmentsDbContext.cs [hunk 2]
```diff
@@ -51,13 +51,33 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
...
+                    details.Property(p => p.TrainingProviderId).HasColumnName("TrainingProviderId");
+                    details.Property(p => p.TrainingProviderName).HasColumnName("TrainingProviderName");
+                    details.OwnsOne(e => e.Course, course =>
+                    {
+                        course.Property(p => p.Name).HasColumnName("CourseName");
+                        course.Property(p => p.Level).HasColumnName("CourseLevel");
+                        course.Property(p => p.Option).HasColumnName("CourseOption");
+                    });
+                });
+
```
Comment (Pauljgraham) [inline src/SFA.DAS.ApprenticeCommitments/Data/Models/ApprenticeCommitmentsDbContext.cs:68]: Nice, first-time I've seen the use of ownsone. Will definitely have conflicts but definitely worthwhile doing this

File: src/SFA.DAS.ApprenticeCommitments/Data/Models/CourseDetails.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            Option = option;
+            PlannedStartDate = plannedStartDate;
+            PlannedEndDate = plannedEndDate;
+        }
+
+        public string Name { get; set; }
+        public int Level { get; set; }
+        public string? Option { get; set; }
+        public DateTime PlannedEndDate { get; set; }
+        public DateTime PlannedStartDate { get; set; }
```
Comment (Pauljgraham) [inline src/SFA.DAS.ApprenticeCommitments/Data/Models/CourseDetails.cs]: Should these be private sets

**Review by Pauljgraham — APPROVED** [review]
I like, approved, one tiny comment about public sets on the course object.

File: src/SFA.DAS.ApprenticeCommitments/Data/Models/CourseDetails.cs (hunk 3 above)
Comment (benagain) [reply src/SFA.DAS.ApprenticeCommitments/Data/Models/CourseDetails.cs]: Yep, good spot, thanks.  I've made them private.

**Pull request merged**